### ISSUE TITLE
Creating Application Secrets

### ISSUE THREAD

**Issue by SquirrelAssassin** [Azure/azure-powershell#4030]
### PowerShell Version
5.1.14393.1198

### Module Version
3.0 

### OS Version
10.0.14393.1198

### Description
Output of Get-AzureRmSubscription not showing subscriptionId

### Script/Steps for Reproduction
run New-AzureRmADAppCredential

### Script/Steps for Reproduction
I was trying to make a secret from PowerShell for a SPN and saw the following: New-AzureRmADAppCredential : Exception of type 'System.Exception' was thrown.

Here is what I did to cause the bug and how to fix it:

created SPN with certificate: New-AzureRmADApplication -DisplayName $adAppName -HomePage $adAppHomePage -IdentifierUris $adAppIdentifierUri -KeyCredentials $keyCredential

created password from portal

tried to create a new password from PowerShell: New-AzureRmADAppCredential -ObjectId 2222-3333-444 -Password 'wakatester'

saw the following error: New-AzureRmADAppCredential : Exception of type 'System.Exception' was thrown.

Debugged and saw the following: "value": "Update to existing credential with KeyId '677dddaaa-4rde-bgt5-853a-2e02bce4c2f1' is not allowed.". This was actually the keyid of the key made from the portal.

to fix I deleted all secrets made from the portal

went back to PowerShell and created new secrets no problem

**Comment by sphibbs** [comment]
Output of Get-AzureRmSubscription not showing subscriptionId was changed to just "ID". Check the migration guide here https://github.com/Azure/azure-powershell/blob/preview/documentation/release-notes/migration-guide.4.0.0.md#breaking-changes-to-profile-cmdlets for details. 

We believe this resolves the issue, if not please reopen add details of your specific issue

**Comment by jamesosb-msft** [comment]
Spent a few hours trying to debug this - I was setting up a certificate for authentication.  I couldn't figure out what was so 'invalid' about the startDate parameter.  I tried using $cer.GetEffectiveDateString(), I tried the actual date string from the cert, various formats, etc.  After suppressing the urge to bang head on keyboard, I tried a date for about 5 minutes from now.  That worked.  It sure would have been nice to know the StartDate isn't for the certificate, but for the credential itself.  And the error message could have been better, saying WHY the date was invalid.
